### DISCUSSION TITLE
Independent Workers

### DISCUSSION THREAD

**Discussion by Muratcol** [taskforcesh/bullmq#1300] [Q&A]
Currently, it is not possible to create a worker without attaching it to a Queue. 

In order to create a more reliable system, I want to create multiple workers and have them pick any available job from any queue.  (Without breaking sequence)
Or another solution, I want to have a backup of a worker without concurrency. If the worker crashes and shuts down, the backup worker should immediately pick up where it left off.

Why is this not possible?

**Comment by manast** [comment]
Why are you asking this?

**Reply by Muratcol** [reply]
@manast  To make the system more reliable. In this way, when any worker crashes, the first available worker will pick up where it left off. Wouldn't it be more convenient for workers to be independent rather than attached to a queue? Or this could be an option?

To be more specific, my concern is what happens to the queue if the worker crash while working on it will another worker continue where it left off? As far as I understood from the documentation that it doesn't work that way.

**Reply by manast** [reply]
When a worker crashes the current active jobs are moved back to the wait status by the QueueScheduler, and then some other worker will pick them up. Here you can read a bit about stalled jobs in a different case, i.e. when the job is so CPU intensive that the job lock cannot be renewed, it is handled the same way as when a worker crashes: https://docs.bullmq.io/guide/workers/stalled-jobs
